### PR TITLE
Renames MessageWithMetadata to MessageContent

### PR DESCRIPTION
**Pull request by conniey**
# Description

* Based on internal conversations, renames this class to MessageContent and into com.azure.
* Updates com.data.schemaregistry.apacheavro to use new package.

.NET: https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/Messaging/MessageContent.cs
JS: https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/schemaregistry/schema-registry-avro/src/models.ts#L7

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/core/azure-core-experimental/src/main/java/com/azure/core/experimental/models/MessageContent.java [hunk 1]
```diff
@@ -10,9 +10,10 @@
  * An abstraction for a message containing a content type along with its data.
  */
 @Fluent
-public class MessageWithMetadata {
+public class MessageContent {
     private BinaryData binaryData;
     private String contentType;
+    private boolean isReadOnly;
```
Comment (srnagar) [inline sdk/core/azure-core-experimental/src/main/java/com/azure/core/experimental/models/MessageContent.java:16]: I don't think we need this in Java.

File: sdk/core/azure-core-experimental/CHANGELOG.md [hunk 2]
```diff
@@ -6,6 +6,8 @@
 
 ### Breaking Changes
 
+- Rename `MessageWithMetadata` to `MessageContent`.
```
Comment (srnagar) [inline sdk/core/azure-core-experimental/CHANGELOG.md:9]: Let's move this to `azure-core` as this has been reviewed in arch board.

File: sdk/schemaregistry/azure-data-schemaregistry-apacheavro/src/main/java/module-info.java [hunk 3]
```diff
@@ -2,8 +2,8 @@
 // Licensed under the MIT License.
 
 module com.azure.data.schemaregistry.apacheavro {
+    requires transitive com.azure.core;
```
Comment (srnagar) [inline sdk/schemaregistry/azure-data-schemaregistry-apacheavro/src/main/java/module-info.java:5]: Is this required? I think transitively requiring `com.azure.data.schemaregistry` should include `com.azure.core` too.

**Review by srnagar — APPROVED** [review]

File: sdk/schemaregistry/azure-data-schemaregistry-apacheavro/src/main/java/module-info.java (hunk 3 above)
Comment (conniey) [reply sdk/schemaregistry/azure-data-schemaregistry-apacheavro/src/main/java/module-info.java:5]: Yeah. I had an error when I didn't. :/

**Pull request merged**